js/ui/file_manager/ui.file_manager.toolbar.js

```diff
@@ -330,20 +351,15 @@ class FileManagerToolbar extends Widget {
     _ensureAvailableCommandsVisible(toolbar, fileItems) {
         toolbar.beginUpdate();
 
-        let groupHasItems = false;
         const items = toolbar.option('items');
 
         items.forEach((item, index) => {
             const itemVisible = item.available;
 
             let showItem = false;
-            if(item.name === 'separator') {
-                showItem = groupHasItems;
-                groupHasItems = false;
-            } else {
+            if(item.name !== 'separator') {
```

Comment:
All separators will have `showItem = false`, so they all will be hidden after this cycle? Maybe we should refactor this code.